WalletWasabi/Blockchain/TransactionOutputs/CoinsRegistry.cs

```diff
@@ -227,12 +227,15 @@ public void SwitchToUnconfirmFromBlock(Height blockHeight)
 	{
 		lock (Lock)
 		{
-			foreach (var coin in AsCoinsViewNoLock().AtBlockHeight(blockHeight))
+			foreach (SmartCoin coin in Coins)
 			{
-				var descendantCoins = DescendantOfNoLock(coin, includeSelf: true);
-				foreach (var toSwitch in descendantCoins)
+				if (coin.Height == blockHeight)
```

Comment:
Rider always suggests to reduce nesting in those cased, and instead write

```
if (coin.Height != blockHeight)
{
    continue;
}
```

I also think that it's slightly better, even if not a big deal